src/android-native/src/main/java/io/liteglue/SQLitePlugin.java

```diff
@@ -496,34 +495,20 @@ private void closeDatabaseNow(String dbname) {
      * @param cbc - JS callback
      */
     private void attachDatabase(String dbName, String dbNameToAttach, String dbAlias, CallbackContext cbc) {
-        SQLiteDatabase currentDb = this.getDatabase(dbName);
-        SQLiteDatabase attachDb = this.getDatabase(dbNameToAttach);
-
-        if (attachDb == null || ! attachDb.isOpen()) {
-            if (cbc != null) cbc.error("Database to attach (" + dbNameToAttach + ") is not open");
-            return;
-        }
-
-        if (currentDb == null || ! currentDb.isOpen()) {
-            if (cbc != null) cbc.error("Database " + dbName + " is not open");
-            return;
-        }
-
-        File dbfile = this.getContext().getDatabasePath(dbNameToAttach);
-        String filePathToAttached = dbfile.getAbsolutePath();
-
-        String stmt = "ATTACH DATABASE '" + filePathToAttached + "' AS " + dbAlias;
-        try {
-            this.executeSqlStatementQuery( currentDb, stmt, new JSONArray(), cbc );
-            // if this previous statement fails, it will throw an exception
-            // otherwise it will never call the success handler because no valid
-            // cursor will be return from rawQuery.
-            // That's why we have to call the success handler here
-            if(cbc != null) cbc.success();
-        }
-        catch(Exception e) {
-            Log.e("attachDatabase", "" + e.getMessage());
-            if(cbc != null) cbc.error("Attach failed");
+        DBRunner r = dbrmap.get(dbNameToAttach);
+        if (r != null) {
+            File dbfile = this.getContext().getDatabasePath(dbNameToAttach);
+            String filePathToAttached = dbfile.getAbsolutePath();
+            String stmt = "ATTACH DATABASE '" + filePathToAttached + "' AS " + dbAlias;
+            // TODO: remove qid it's hardcoded in js to be 1111 always anyway
+            DBQuery query = new DBQuery(new String[]{stmt}, new String[]{"1111"}, new JSONArray[]{new JSONArray()}, cbc);
+            try {
+                r.q.put(query);
```

Comment:
correct threading model.
We don't execute DB queries in the ReactNative thread.
It's violation of the module threading model and could cause issues hard to debug.